### PR TITLE
external ranking expressions

### PR DESCRIPTION
**Pull request by havardpe**
loaded from potentially compressed files

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@geirst please review
@baldersheim FYI

### CODE REVIEW

File: searchcore/src/vespa/searchcore/proton/server/documentdbconfigmanager.cpp [hunk 1]
```diff
@@ -344,6 +350,31 @@ DocumentDBConfigManager::update(const ConfigSnapshot &snapshot)
...
+        RankingExpressionsConfigSP newRankingExpressionsConfig = RankingExpressionsConfigSP(
+            snapshot.getConfig<RankingExpressionsConfig>(_configId));
+        const vespalib::string &spec = _bootstrapConfig->getFiledistributorrpcConfig().connectionspec;
+        RankingExpressions expressions;
+        if (spec != "") {
+            config::RpcFileAcquirer fileAcquirer(spec);
+            vespalib::TimeBox timeBox(5*60, 5);
+            for (const RankingExpressionsConfig::Expression &rc : newRankingExpressionsConfig->expression) {
+                vespalib::string filePath;
+                LOG(info, "Waiting for file acquirer (name='%s', ref='%s')",
```
Comment (geirst) [inline searchcore/src/vespa/searchcore/proton/server/documentdbconfigmanager.cpp:363]: Lines 363 to 372 are almost identical to similar code handling ranking constants and onnx models. Consider adding a helper function to reduce code duplication.

**Review by geirst — APPROVED** [review]
👍

File: searchcore/src/apps/verify_ranksetup/verify_ranksetup.cpp [hunk 2]
```diff
@@ -57,6 +61,19 @@ std::optional<vespalib::string> get_file(const vespalib::string &ref, const Veri
     return std::nullopt;
 }
 
+RankingExpressions make_expressions(const RankingExpressionsConfig &expressionsCfg, const VerifyRanksetupConfig &myCfg) {
+    RankingExpressions expressions;
+    for (const auto &entry: expressionsCfg.expression) {
+        if (auto file = get_file(entry.fileref, myCfg)) {
+            expressions.add(entry.name, file.value());
+        } else {
+            LOG(warning, "could not find file for ranking expression '%s' (ref:'%s')\n",
```
[the remark this answers is not in the document]
Comment (havardpe) [reply searchcore/src/apps/verify_ranksetup/verify_ranksetup.cpp]: fixed

File: searchcore/src/vespa/searchcore/proton/server/documentdbconfigmanager.cpp (hunk 1 above)
Comment (havardpe) [reply searchcore/src/vespa/searchcore/proton/server/documentdbconfigmanager.cpp]: fixed

**Review by baldersheim — APPROVED** [review]

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**